### DISCUSSION TITLE
Retrieve gene list of ambient profile

### DISCUSSION THREAD

**Discussion by carolienv** [campbio/decontX#12] [Q&A] [answered]
Is there a way to retrieve a list of genes that contribute most to the ambient RNA profile?

**Comment by joshua-d-campbell** [comment, marked as answer]
Hi @carolienv, thanks for using our tool. Yes, when running the `decontX` function for scRNA-seq, you can get the parameters back out of the metadata (see code below). "Eta" is a "gene by cell population matrix". The higher the probability the higher the gene contributes to the ambient profile. Note that many of the highly expressed housekeeping genes are highly ranked in the ambient profiles as well. To find the genes that are enriched in the ambient profile, you may want to compare it to the cell profile. You can get the cell estimates from "Phi" which is also a "gene by cell population" matrix. Here is some code to do this:

```
library(celda)
library(singleCellTK)
sce <- singleCellTK::importExampleData("pbmc3k").  # Get PBMC example from singleCellTK
sce <- decontX(sce). # Run decontX with default parameters. 

# Get the ambient profiles
# If you give it the "batch" parameter, then you will have to change "all_cells" in the path below to the name of each batch
eta <- metadata(sce)$decontX$estimates$all_cells$eta
rownames(eta) <- rownames(sce)

# Get the cell population profile
phi <- metadata(sce)$decontX$estimates$all_cells$phi
rownames(phi) <- rownames(sce)

# Visualize differences in cell profile vs. ambient profile for cell cluster 1
# The points off the diagonal towards Eta are genes that are enriched in the contamination distribution. 
plot(phi[,1], eta[,1])
```

This is an interesting question, so I am going to move it to a Discussion.

**Comment by sjessa** [comment]
Hi @joshua-d-campbell, thanks for all this work on the package! I had a similar question - I've explored the eta distributions in my decontX output, and I'm wondering if it's also possible to extract, for each cluster $k$, the weights in $\eta_k$ i.e. the weights $w_k'$ (for $k':k'\neq k$) on all the native distributions that contribute to the contamination of $k$?
